test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
+		assert.ok( /barf/.test( msg ), "Message: " + msg );
+	};
+	jQuery.when(
+		defer.then( function() {
+			// Should get an error
+			jQuery.barf();
```

Comment:
Instead of calling `jQuery.barf`, we could make a direct call to the assertion here, the `assert.expect` is enough.